src/components/application_manager/src/commands/mobile/system_request.cc

```diff
@@ -224,63 +592,23 @@ void SystemRequest::on_event(const event_engine::Event& event) {
 }
 
 bool SystemRequest::ValidateQueryAppData(
-    const smart_objects::SmartObject& data) const  {
+    const smart_objects::SmartObject& data) const {
   if (!data.isValid()) {
-    LOG4CXX_ERROR(logger_, "QueryApps response is not valid.");
+    LOG4CXX_ERROR(logger_,
+                  kQeeryAppsValidationFailedPrefix
+                      << "QueryApps response is not valid.");
     return false;
   }
   if (!data.keyExists(json::response)) {
     LOG4CXX_ERROR(logger_,
-                  "QueryApps response does not contain '"
-                  << json::response << "' parameter.");
+                  kQeeryAppsValidationFailedPrefix
+                      << "QueryApps response does not contain '"
+                      << json::response << "' parameter.");
     return false;
   }
-  smart_objects::SmartArray* obj_array = data[json::response].asArray();
-  if (NULL == obj_array) {
-    return false;
-  }
-
-  const std::size_t arr_size(obj_array->size());
-  for (std::size_t idx = 0; idx < arr_size; ++idx) {
-    const smart_objects::SmartObject& app_data = (*obj_array)[idx];
-    if (!app_data.isValid()) {
-      LOG4CXX_ERROR(logger_, "Wrong application data in json file.");
-      continue;
-    }
-    std::string os_type;
-    if (app_data.keyExists(json::ios)) {
-      os_type = json::ios;
-      if (!app_data[os_type].keyExists(json::urlScheme)) {
-        LOG4CXX_ERROR(logger_, "Can't find URL scheme in json file.");
-        continue;
-      }
-    } else if (app_data.keyExists(json::android)) {
-      os_type = json::android;
-      if (!app_data[os_type].keyExists(json::packageName)) {
-        LOG4CXX_ERROR(logger_, "Can't find package name in json file.");
-        continue;
-      }
-    }
-
-    if (os_type.empty()) {
-      LOG4CXX_ERROR(logger_, "Can't find mobile OS type in json file.");
-      continue;
-    }
-
-    if (!app_data.keyExists(json::appId)) {
-      LOG4CXX_ERROR(logger_, "Can't find app ID in json file.");
-      continue;
-    }
-
-    if (!app_data.keyExists(json::name)) {
-      LOG4CXX_ERROR(logger_, "Can't find app name in json file.");
-      continue;
-    }
-
-    return true;
-  }
 
-  return false;
+  QueryAppsDataValidatorTemplate<smart_objects::SmartObject> validator(data);
```

Comment:
@okoreniak, since this validator can work with SmartObject **only**, I see no reason in template. I see no harm in additional file and splitting declaration and impl. Also templates makes more pressure on the compiler. So I propose to turn into simple class.